cppapi/client/event.cpp

```diff
@@ -1822,6 +1822,54 @@ int EventConsumer::connect_event(DeviceProxy *device,
 	return ret_event_id;
 }
 
+string EventConsumer::get_client_attribute_name(const string &local_callback_key, const vector<string> &filters)
+{
+    if(filters.size() == 1)
+        return filters[0];// here filters[0] is expected to be user defined attribute name
+
+
+	size_t pos = local_callback_key.rfind('.');//remove event_type e.g. .idl5_change
+	return local_callback_key.substr(0, pos);
+}
+
+void Tango::EventConsumer::initialize_received_from_admin(const Tango::DevVarLongStringArray *dvlsa,
+														  const string &local_callback_key,
+														  const string &adm_name,
+														  bool device_from_env_var)
+{
+    long server_tango_lib_ver = dvlsa->lvalue[0];
+
+    if (server_tango_lib_ver >= 930)
+    {
+        received_from_admin.event_name = (dvlsa->svalue[dvlsa->svalue.length() - 2]);
+    }
+    else
+    {
+        received_from_admin.event_name = local_callback_key;
+    }
+
+    if (server_tango_lib_ver >= 930)
```

Comment:
I think this can be grouped with the previous if which is on line 1842

Reply:
I intentionally split them, because these are two distinct actions. Keep them separate will help in the future refactoing